draco/analysis/delay.py

```diff
@@ -718,6 +718,212 @@ def process(self, ss: FreqContainerType) -> containers.DelaySpectrum:
         return delay_spec
 
 
+class DelaySpectrumWienerBase(task.SingleTask):
+    """Calculate the delay spectrum for each RA and DEC of any container with a frequency axis.
+
+    The spectrum is calculated by Wiener filtering method given the delay power spectrum
+    at each DEC/baselines [https://arxiv.org/abs/2202.01242 see:- eqn A6].
+
+    The delay spectrum output is indexed by  `baseline` and 'sample' axis. The baseline axis is the
+    composite axis of all the axis (DEC, pol and beam axes) in the container except the 'sample' and frequency axis.
+    The 'sample' axis is the average axis over which the average is taken to estimate the delay power spectrum.
+    These constituent axes are included in the index map,  which are ['beam', 'pol', 'sample', 'el'], and their
+    order is given by the `baseline_axes` attribute.
+
+    Attributes
+    ----------
+
+    dataset : str
+        Calculate the delay spectrum of this dataset (e.g., "vis", "map", "beam").
+    average_axis : str
+        Name of the axis to take the average over.
```

Comment:
Unlike in the Gibbs delay power spectrum estimator, no average is taken in this task, so a different name should probably be used here